azurerm/helpers/azure/resource_group.go

```diff
@@ -61,12 +61,12 @@ func SchemaResourceGroupNameSetOptional() *schema.Schema {
 		Optional: true,
 		Elem: &schema.Schema{
 			Type:         schema.TypeString,
-			ValidateFunc: validateResourceGroupName,
+			ValidateFunc: ValidateResourceGroupName,
 		},
 	}
 }
 
-func validateResourceGroupName(v interface{}, k string) (warnings []string, errors []error) {
+func ValidateResourceGroupName(v interface{}, k string) (warnings []string, errors []error) {
```

Comment:
rather then exporting, could we just add a new schema function `SchemaResourceGroupNameOptional` and use that instead?

Reply:
updated